test/bin/scenario.sh

```diff
@@ -65,7 +65,13 @@ run_command_on_vm() {
     local -r ip=$(get_vm_property "${vmname}" ip)
     local -r ssh_port=$(get_vm_property "${vmname}" ssh_port)
 
-    ssh "redhat@${ip}" -p "${ssh_port}" -t "${command}"
+    local term_opt=""
+    if [ -t 0 ] ; then
+        # Allocate pseudo-terminal for SSH commands when stdin is a terminal
+        # Necessary in devenv for entering input i.e. system registration, etc.
+        term_opt="-t"
+    fi
+    ssh "redhat@${ip}" -p "${ssh_port}" "${term_opt}" "${command}"
```

Comment:
What happens here if term_opt is not set and ssh gets an empty argument? I'd expect to not quote the variable here so that the empty variable would be transparently ignored.

Reply:
Empty arguments are usually ignored by commands, it's like a no-op
```
$ args=""
$ ssh microshift@microshift-dev-rhel92 -p 22 "${args}" "hostname"
microshift-dev-rhel92

$ args="-t"
$ ssh microshift@microshift-dev-rhel92 -p 22 "${args}" "hostname"
microshift-dev-rhel92
Connection to microshift-dev-rhel92 closed.
```

Reply:
In any case, I will fix it not to raise questions.

Reply:
It does not even need disabling the warning because we're manipulating constants.
```
    # shellcheck disable=SC2086
```